### PR TITLE
RBAC - read-only user guide and reference doc refactoring

### LINKED ISSUE #245: [Guide] How to create a read-only user with RBAC

### LINKED ISSUE #267: [Reference] RBAC
The RBAC reference documentation needs a refactoring!

### PR DESCRIPTION
**Pull request by mercul3s**
Closes #245 
Closes #267

### CODE REVIEW

File: content/sensu-core/2.0/guides/create-read-only-user.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+version: "2.0"
+product: "Sensu Core"
+platformContent: False
+menu: 
+  sensu-core-2.0:
+    parent: guides
+---
+
+## What is RBAC?
+**Role-Based Access Control** (RBAC) is Sensu's local user management system. RBAC currently supports the management of users and permissions with Organizations, Environments, Roles, and Users. Support for external directories (such as LDAP) will be available in a future release.
```
Comment (palourde) [inline content/sensu-core/2.0/guides/create-read-only-user.md]: As per [Matt suggestion](https://github.com/sensu/sensu-docs-site/pull/223#discussion_r171637928), I think we should decapitalize `Organizations, Environments, Roles, and Users` and bold these terms instead. What do you think?

File: content/sensu-core/2.0/guides/create-read-only-user.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Why use RBAC
+RBAC allows you to exercise fine-grained control over how Sensu users interact 
+with Sensu resources. Using RBAC rules, you can easily achieve **multitenancy** 
+so users only have access to resources within their own organization and environment. 
+
+## How to create a Read Only user
+The initial installation of Sensu includes a `default` environment, included in the `default` organization, and an `admin` user with full permissions to create, modify, or delete resouces within Sensu. Using `sensuctl`, you can create new roles for users that give as much or as little access as you see fit. The purpose of this guide is to help you create a new user with read-only access to resources within the `default` organization and environments.
```
Comment (palourde) [inline content/sensu-core/2.0/guides/create-read-only-user.md]: s/environments/environment/

File: content/sensu-core/2.0/reference/rbac.md [hunk 3]
```diff
@@ -10,117 +10,200 @@ menu:
     parent: reference
```
Comment (palourde) [inline content/sensu-core/2.0/reference/rbac.md]: Can you switch `platformContent` to `false`?

File: content/sensu-core/2.0/reference/rbac.md [hunk 4]
```diff
@@ -10,117 +10,200 @@ menu:
...
-# Role-Based Access Control (RBAC) and Multitenancy in Sensu 2.0
-
-Sensu's **Role-Based Access Control** (RBAC) allows you to exercise fine-grained
-control over how Sensu users interact with Sensu resources. Using RBAC
-rules, you can easily achieve **multitenancy** so users only have access to
-resources within their own organization and environment.
-
-## Overview
+## How does Role Based Access Control work?
+Sensu RBAC allows management and access of users and resources based on a heirarchy of Organizations, Environments, Roles, Rules, and Users. Each `environment` belongs to only one `organization`, and each `resource` (user, check, asset, etc) belongs to only one `environment`. A Sensu installation can have multiple organizations, each with their own set of environments and resources belonging to them.
```
Comment (palourde) [inline content/sensu-core/2.0/reference/rbac.md]: Same thing as above regarding capitalization!

File: content/sensu-core/2.0/reference/rbac.md [hunk 5]
```diff
@@ -10,117 +10,200 @@ menu:
...
+        "organization": "*",
+        "permissions": [
+            "create",
+            "read",
+            "update",
+            "delete"        
+        ]
+    } 
+]{{</highlight>}}
+
```
Comment (palourde) [inline content/sensu-core/2.0/reference/rbac.md]: Could you add a full example here? Nothing complex, the `read-only` role could be perfect!

**Review by palourde — CHANGES_REQUESTED** [review]
Amazing work 👏 👏 👏 

Just some small details, otherwise it's perfect!

File: content/sensu-core/2.0/guides/create-read-only-user.md (hunk 1 above)
Comment (mercul3s) [reply content/sensu-core/2.0/guides/create-read-only-user.md]: That's fine with me! I'll update it.

File: content/sensu-core/2.0/reference/rbac.md (hunk 3 above)
Comment (mercul3s) [reply content/sensu-core/2.0/reference/rbac.md]: Sure. That brings up a question that's been on the back of my mind - do we have a template for what should be in the front matter? I've been using hugo to generate and adding in the menu tree stuff from other docs, but haven't been really paying attention to what *should* be there at a minimum or what those attributes do 😳

File: content/sensu-core/2.0/reference/rbac.md (hunk 5 above)
Comment (mercul3s) [reply content/sensu-core/2.0/reference/rbac.md]: Do you mean for rule or role?

Comment (mercul3s) [reply content/sensu-core/2.0/reference/rbac.md]: Oh nvm, couldn't see which line the comment was on until I switched to the inline view *shakes fist at github*

**Comment by mercul3s** [comment]
@palourde I think I've addressed all your comments, ready for another round of review.

File: content/sensu-core/2.0/reference/rbac.md (hunk 3 above)
Comment (palourde) [reply content/sensu-core/2.0/reference/rbac.md]: I know we have these templates now for Sensu 1.0: https://github.com/sensu/sensu-docs-site/blob/master/content/sensu-core/0.29/files/sensu-doc-template.md

But more specifically, `platformContent` is used to generate a dropdown button to switch the platform (i.e., rhel, ubuntu, windows) used so we can create sections specific to each platform. The [installation](https://github.com/sensu/sensu-docs-site/blob/master/content/sensu-core/2.0/getting-started/installation-and-configuration.md) guide is a good example. I don't believe we have any documentation on that though.

**Review by palourde — APPROVED** [review]

File: content/sensu-core/2.0/reference/rbac.md (hunk 3 above)
Comment (mercul3s) [reply content/sensu-core/2.0/reference/rbac.md]: Oh gotcha, that makes total sense.

**Pull request merged**